knowledge_runtime/start.sh

```
@@ -0,0 +1,290 @@
+    exit 1
+fi
+
+# Create virtual environment if needed
+if [ ! -d ".venv" ]; then
+    echo "Creating virtual environment..."
+    uv venv
+fi
+
+# Sync dependencies
+uv sync
+if [ $? -ne 0 ]; then
```

Shellcheck (0.11.0)

[style] Check exit code directly with e.g. 'if ! mycmd;', not indirectly with $?.

(SC2181)